Dockerfile

```diff
@@ -3,26 +3,59 @@ FROM runatlantis/atlantis:latest
 RUN apk add unzip make curl
 
 # TODO: verify GPG
-ARG TERRAGRUNT_VERSION='0.21.7'
-ARG DEFAULT_TERRAFORM_VERSION=0.12.29
-ENV DEFAULT_TERRAFORM_VERSION=$DEFAULT_TERRAFORM_VERSION
+ARG TERRAGRUNT_VERSION=0.21.7
+ENV DEFAULT_TERRAGRUNT_VERSION=$TERRAGRUNT_VERSION
+ARG TERRAFORM_VERSION=0.12.29
+ENV DEFAULT_TERRAFORM_VERSION=$TERRAFORM_VERSION
 
-RUN  \
-  rm -rf /usr/local/bin/terraform \
-  && ln -s /usr/local/bin/tf/versions/${DEFAULT_TERRAFORM_VERSION}/terraform /usr/local/bin/terraform
+RUN set -eux \
+    && if [ "${TERRAFORM_VERSION}" = "latest" ]; then \
+          DEFAULT_TERRAFORM_VERSION="$( curl -sS https://releases.hashicorp.com/terraform/ \
+                  | tac | tac \
+                  | grep -Eo '/[.0-9]+/' \
+                  | grep -Eo '[.0-9]+' \
+                  | sort -V \
+                  | tail -1 )"; \
+    else \
+        DEFAULT_TERRAFORM_VERSION="$( curl -sS https://releases.hashicorp.com/terraform/ \
+			| tac | tac \
+			| grep -Eo "/${TERRAFORM_VERSION}\.[.0-9]+/" \
+			| grep -Eo '[.0-9]+' \
+			| sort -V \
+			| tail -1 )"; \
+    fi \
+    && rm -rf /usr/local/bin/terraform \
+    && ln -s /usr/local/bin/tf/versions/${DEFAULT_TERRAFORM_VERSION}/terraform /usr/local/bin/terraform
 
-RUN \
-# Download the binary and checksum
- curl -OLSs "https://github.com/gruntwork-io/terragrunt/releases/download/v${TERRAGRUNT_VERSION}/terragrunt_linux_amd64" \
- && curl -OLSs "https://github.com/gruntwork-io/terragrunt/releases/download/v${TERRAGRUNT_VERSION}/SHA256SUMS" \
-# Verify the SHASUM matches the binary.
- && grep terragrunt_linux_amd64 SHA256SUMS > terragrunt_linux_amd64_SHA256SUMS \
- && sha256sum -c terragrunt_linux_amd64_SHA256SUMS \
- && chmod a+x terragrunt_linux_amd64 \
- && mv terragrunt_linux_amd64 /usr/local/bin/terragrunt \
-# Clean up
- && rm -rf "SHA256SUMS" \
-   "terragrunt_linux_amd64_SHA256SUMS"
+RUN set -eux \
+    && if [ "${TERRAGRUNT_VERSION}" = "latest" ]; then \
+          DEFAULT_TERRAGRUNT_VERSION="$( curl -L -sS https://github.com/gruntwork-io/terragrunt/releases \
+                  | tac | tac \
+                  | grep -Eo '/v[.0-9]+/' \
+                  | grep -Eo '[.0-9]+' \
+                  | sort -u \
+                  | sort -V \
+                  | tail -1 )"; \
+    elif [ "${TERRAGRUNT_VERSION}" = "0.21.7" ]; then \
+        DEFAULT_TERRAGRUNT_VERSION=$TERRAGRUNT_VERSION; \
```

Comment:
Why do we need a specific case for a patch level version? Isn't that handled by the `else` below?
```suggestion
```

Reply:
This is because of the limitation we have regarding terragrunt version, the latest version from branch 0.21 its 0.21.13
For some reason we have been only using 0.21.7, maybe we can bypass this and jump to 0.21.13